src/components/BarSeries.tsx

```diff
@@ -37,22 +37,26 @@ export function BarSeries(props: BarSeriesProps) {
       hidden: !displayMarker,
     };
   }, [color, displayMarker, figure]);
-
+  const visibility = useMemo(() => {
+    const value = legendState.labels.find((label) => label.id === id);
+    return value ? value.visibility : true;
+  }, [id, legendState.labels]);
```

Comment:
This same code is used in all series components. Please refactor to a custom hook. `const isVisible = useIsSeriesVisible(id)`